src/main/java/com/ldtteam/blockout/views/TabSet.java

```diff
@@ -0,0 +1,271 @@
+package com.ldtteam.blockout.views;
+
+import com.ldtteam.blockout.Alignment;
+import com.ldtteam.blockout.Pane;
+import com.ldtteam.blockout.PaneParams;
+import com.ldtteam.blockout.controls.Image;
+import com.ldtteam.blockout.controls.Text;
+import com.mojang.blaze3d.matrix.MatrixStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.Tuple;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Random;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Similar to a switch view, the tab set renders the selected view,
+ * allowing for tabs to be placed for user's selection
+ */
+public class TabSet extends View
+{
+    private static final Random RANDOM = new Random();
+
+    protected boolean vertical;
+    protected int spacing;
+    protected int overlap;
+    protected int offset;
+    protected int tabWidth;
+    protected int tabHeight;
+
+    protected Tab selectedTab;
+
+    /** The group of tab icons */
+    protected int childPadding;
+    protected final View         tabs;
+    protected ResourceLocation[] tabIcons = new ResourceLocation[0];
+
+    public TabSet(PaneParams params)
+    {
+        super(params);
+        vertical = params.getBoolean("vertical", false);
+        overlap = params.getInteger("overlap", 0);
+
+        tabWidth = params.getInteger("tabwidth", 100);
+        tabHeight = params.getInteger("tabheight", 20);
+        offset = params.getInteger("taboffset", 0);
+
+        childPadding = padding;
+        padding = 0;
```

Comment:
huh, probably connected to comment below, but why don't you just use padding of this element?

Reply:
Planned use case. Some elements in the TabView may wish to ignore padding, such as images for backgrounds.